subsys/bluetooth/audio/ascs.c

```diff
@@ -2888,4 +2888,145 @@ void bt_ascs_cleanup(void)
+						     BT_BAP_ASCS_REASON_NONE);
+	int err;
+
+	ep = stream->ep;
+
+	if (unicast_server_cb != NULL &&
+		unicast_server_cb->reconfig != NULL) {
+		err = unicast_server_cb->reconfig(stream, ep->dir, codec,
+						  &ep->qos_pref, &rsp);
+	} else {
+		err = -ENOTSUP;
+	}
+
+	if (err != 0) {
+		return err;
+	}
+
+	(void)memcpy(&ep->codec, &codec, sizeof(codec));
+
+	ascs_ep_set_state(ep, BT_BAP_EP_STATE_CODEC_CONFIGURED);
+
+	return 0;
+}
+
+int bt_ascs_ase_start(struct bt_bap_stream *stream)
+{
+	struct bt_bap_ep *ep = stream->ep;
+
+	if (ep->dir != BT_AUDIO_DIR_SINK) {
+		LOG_DBG("Invalid operation for stream %p with dir %u",
+			stream, ep->dir);
+
+		return -EINVAL;
+	}
+
+	/* If ISO is connected to go streaming state,
+	 * else wait for ISO to be connected
+	 */
+	if (ep->iso->chan.state == BT_ISO_STATE_CONNECTED) {
+		ascs_ep_set_state(ep, BT_BAP_EP_STATE_STREAMING);
+	} else {
+		ep->receiver_ready = true;
+	}
+
+	return 0;
+}
+
+int bt_ascs_ase_metadata(struct bt_bap_stream *stream, struct bt_codec_data meta[],
+			 size_t meta_count)
+{
+	struct bt_bap_ep *ep;
+	struct bt_bap_ascs_rsp rsp = BT_BAP_ASCS_RSP(BT_BAP_ASCS_RSP_CODE_SUCCESS,
+						     BT_BAP_ASCS_REASON_NONE);
+	int err;
+
+
+	if (unicast_server_cb != NULL && unicast_server_cb->metadata != NULL) {
+		err = unicast_server_cb->metadata(stream, meta, meta_count, &rsp);
+	} else {
+		err = -ENOTSUP;
+	}
+
+	ep = stream->ep;
+	for (size_t i = 0U; i < meta_count; i++) {
+		(void)memcpy(&ep->codec.meta[i], &meta[i],
+			     sizeof(ep->codec.meta[i]));
+	}
+
+	if (err) {
+		LOG_ERR("Metadata failed: err %d, code %u, reason %u", err, rsp.code, rsp.reason);
+		return err;
+	}
+
+	/* Set the state to the same state to trigger the notifications */
+	ascs_ep_set_state(ep, ep->status.state);
+
+	return 0;
+}
+
+int bt_ascs_ase_disable(struct bt_bap_stream *stream)
+{
+	struct bt_bap_ep *ep;
+	struct bt_bap_ascs_rsp rsp = BT_BAP_ASCS_RSP(BT_BAP_ASCS_RSP_CODE_SUCCESS,
+						     BT_BAP_ASCS_REASON_NONE);
+	int err;
+
+	if (unicast_server_cb != NULL && unicast_server_cb->disable != NULL) {
+		err = unicast_server_cb->disable(stream, &rsp);
+	} else {
+		err = -ENOTSUP;
+	}
+
+	if (err != 0) {
+		LOG_ERR("Disable failed: err %d, code %u, reason %u", err, rsp.code, rsp.reason);
+		return err;
+	}
+
+	ep = stream->ep;
+
+	/* The ASE state machine goes into different states from this operation
+	 * based on whether it is a source or a sink ASE.
+	 */
+	if (ep->dir == BT_AUDIO_DIR_SOURCE) {
+		ascs_ep_set_state(ep, BT_BAP_EP_STATE_DISABLING);
+	} else {
+		ascs_ep_set_state(ep, BT_BAP_EP_STATE_QOS_CONFIGURED);
+	}
+
+	return 0;
+}
+
+int bt_ascs_ase_release(struct bt_bap_stream *stream)
+{
+	struct bt_bap_ascs_rsp rsp = BT_BAP_ASCS_RSP(BT_BAP_ASCS_RSP_CODE_SUCCESS,
+						     BT_BAP_ASCS_REASON_NONE);
+	int err;
+
+	if (unicast_server_cb != NULL && unicast_server_cb->release != NULL) {
+		err = unicast_server_cb->release(stream, &rsp);
+	} else {
+		err = -ENOTSUP;
+	}
```

Comment:
Since these callbacks are (due to their naming) more relevant for the unicast server implementation than the ASCS implement, perhaps it would make sense to keep the callbacks in `bap_unicast_server.c`?

Reply:
This makes sense